run/helm/snap/templates/config/_snapteld.tpl

```diff
@@ -0,0 +1,99 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# log_level for the snap daemon. Supported values are
+# 1 - Debug, 2 - Info, 3 - Warning, 4 - Error, 5 - Fatal.
+# Default value is 3.
+log_level: {{ .Values.config.log_level }}
+
+# log_path sets the path for logs for the snap daemon. By
+# default snapteld prints all logs to stdout. Any provided
+# path will send snapteld logs to a file called snapteld.log in
+# the provided directory.
+log_path: {{ .Values.config.log_path }}
+
+# Gomaxprocs sets the number of cores to use on the system
+# for snapteld to use. Default for gomaxprocs is 1
+# gomaxprocs: 1
+
+# Control sections for configuration settings for the plugin
+# control module of snapteld.
+control:
+  # auto_discover_path sets a directory to auto load plugins on the start
+  # of the snap daemon
+  auto_discover_path: {{ .Values.config.autodiscover_path }} 
+
+  # cache_expiration sets the time interval for the plugin cache to use before
+  # expiring collection results from collect plugins. Default value is 500ms
+  # cache_expiration: 500ms
+
+  # max_running_plugins sets the size of the available plugin pool for each
+  # plugin loaded in the system. Default value is 3
+  # max_running_plugins: 3
+
+  # keyring_paths sets the directory(s) to search for keyring files for signed
+  # plugins. This can be a comma separated list of directories
+  # keyring_paths: /etc/snap/keyrings
+
+  # plugin_trust_level sets the plugin trust level for snapteld. The default state
+  # for plugin trust level is enabled (1). When enabled, only signed plugins that can
+  # be verified will be loaded into snapteld. Signatures are verifed from
+  # keyring files specided in keyring_path. Plugin trust can be disabled (0) which
+  # will allow loading of all plugins whether signed or not. The warning state allows
+  # for loading of signed and unsigned plugins. Warning messages will be displayed if
+  # an unsigned plugin is loaded. Any signed plugins that can not be verified will
+  # not be loaded. Valid values are 0 - Off, 1 - Enabled, 2 - Warning
+  plugin_trust_level: 0
+
+  # plugins section contains plugin config settings that will be applied for
+  # plugins across tasks.
+  # plugins:
+
+# scheduler configuration settings contains all settings for scheduler
+# module
+# scheduler:
+  # work_manager_queue_size sets the size of the worker queue inside snapteld scheduler.
+  # Default value is 25.
+  # work_manager_queue_size: 25
+
+  # work_manager_pool_size sets the size of the worker pool inside snapteld scheduler.
+  # Default value is 4.
+  # work_manager_pool_size: 4
+
+# rest sections contains all the configuration items for the REST API server.
+# restapi:
+  # enable controls enabling or disabling the REST API for snapteld. Default value is enabled.
```

Comment:
just to clarify, does this deploy snap with unencrypted rest with no auth? Should that be the default?